staker/stakerapp.go

```diff
@@ -1454,6 +1515,61 @@ func (app *App) handleStakingCommands() {
 				)
 			}
 			app.logStakingEventProcessed(cmd)
+
+		case cmd := <-app.migrateStakingCmd:
+			stkTxHash := cmd.notifierTx.Tx.TxHash()
+			stakingParams, err := app.babylonClient.Params()
+			if err != nil {
+				cmd.errChan <- err
+				continue
+			}
+
+			bestBlockHeight := app.currentBestBlockHeight.Load()
+			if err := app.waitForStakingTransactionConfirmation(
+				&stkTxHash,
+				cmd.parsedStakingTx.StakingOutput.PkScript,
+				stakingParams.ConfirmationTimeBlocks,
+				bestBlockHeight,
+			); err != nil {
+				cmd.errChan <- err
+				continue
+			}
+
+			if err := app.txTracker.AddTransactionSentToBTC(
+				cmd.notifierTx.Tx,
+				uint32(cmd.parsedStakingTx.StakingOutputIdx),
+				cmd.parsedStakingTx.OpReturnData.StakingTime,
+				[]*btcec.PublicKey{cmd.parsedStakingTx.OpReturnData.FinalityProviderPublicKey.PubKey},
+				babylonPopToDBPop(cmd.pop),
+				cmd.stakerAddr,
+			); err != nil {
+				app.logger.WithError(err).Info("err to set tx as sent on BTC")
+				cmd.errChan <- err
+				continue
+			}
+
+			// eventually tx is send to babylon
+			storedTx, err := app.waitForTrackedTransactionState(stkTxHash, proto.TransactionState_SENT_TO_BABYLON, time.Second, 20)
+			if err != nil {
+				utils.PushOrQuit(
+					cmd.errChan,
+					err,
+					app.quit,
+				)
+				app.logger.WithFields(logrus.Fields{
+					"stakingTxHash": stkTxHash,
+				}).Debugf("BTC delegation waited for too long to become active, check the status manually")
+				continue
+			}
```

Comment:
not fully convinced it is a good idea to retrieve the consumer delegation tx hash this way